### PR TITLE
Filter out empty dfs in gds.graph.construct

### PR DESCRIPTION
**Pull request by FlorentinD**

### CODE REVIEW

**Comment by FlorentinD** [comment]
fixes #460

**Comment by FlorentinD** [comment]
[#460 Neo.ClientError.Procedure.ProcedureCallFailed when constructing a GDS graph object from DataFrame](https://trello.com/c/BFfVTaLl/8611-460-neoclienterrorprocedureprocedurecallfailed-when-constructing-a-gds-graph-object-from-dataframe)

**Review by IoannisPanagiotas — APPROVED** [review]
Nice !

File: graphdatascience/graph/graph_proc_runner.py [hunk 1]
```diff
@@ -55,13 +55,17 @@ def construct(
...
+        relationships: Union[DataFrame, List[DataFrame]] = None,
         concurrency: int = 4,
         undirected_relationship_types: Optional[List[str]] = None,
     ) -> Graph:
         nodes = nodes if isinstance(nodes, List) else [nodes]
         relationships = relationships if isinstance(relationships, List) else [relationships]
 
+        # Filter empty dataframes
+        nodes = [df for df in nodes if not df.empty]
+        relationships = [df for df in relationships if not df is None and not df.empty]
```
Comment (Mats-SX) [inline graphdatascience/graph/graph_proc_runner.py:67]: I think you can simplify a little bit 

```suggestion
        relationships = [df for df in relationships if df and not df.empty]
```

File: graphdatascience/tests/unit/test_graph_construct.py [hunk 2]
```diff
@@ -262,3 +262,7 @@ def test_graph_alpha_construct_backward_compat(runner: CollectingQueryRunner, gd
 
     with pytest.warns(DeprecationWarning):
         gds.alpha.graph.construct("hello", nodes, relationships, concurrency=2)
+
+def test_nodes_graph(gds: GraphDataScience) -> None:
+    gds.graph.construct("hello", nodes = DataFrame({"nodeId": [0, 1]}), concurrency=2)
+    gds.graph.construct("hello", nodes = DataFrame({"nodeId": [0, 1]}), relationships=[], concurrency=2)
```
Comment (Mats-SX) [inline graphdatascience/tests/unit/test_graph_construct.py:268]: no assertions?

File: graphdatascience/graph/graph_proc_runner.py (hunk 1 above)
Comment (FlorentinD) [reply graphdatascience/graph/graph_proc_runner.py:67]: not working unfortunatly as the boolean value of df is ambigous 🙈

Comment (Mats-SX) [reply graphdatascience/graph/graph_proc_runner.py:67]: okay fair enough

File: graphdatascience/tests/integration/test_graph_construct.py [hunk 3]
```diff
@@ -477,6 +476,27 @@ def test_graph_alpha_construct_backward_compat_without_arrow(gds_without_arrow:
         gds_without_arrow.alpha.graph.construct("hello", nodes, relationships)
 
 
+def test_nodes_only_without__arrow(gds_without_arrow: GraphDataScience) -> None:
+    nodes = DataFrame({"nodeId": [0], "labels": ["person"]})
+    relationships = DataFrame({"sourceNodeId": [], "targetNodeId": [], "relationshipType": []})
+
+    G = gds_without_arrow.graph.construct(graph_name="my_graph", nodes=nodes, relationships=relationships)
```
Comment (Mats-SX) [inline graphdatascience/tests/integration/test_graph_construct.py:483]: this produces a warning that we must catch

**Pull request merged**